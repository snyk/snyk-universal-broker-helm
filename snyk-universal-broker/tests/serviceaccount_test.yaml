# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Service Account configurations
templates:
  - templates/statefulset.yaml
  - templates/serviceaccount.yaml
values:
  - ../values.yaml
  - fixtures/default_values.yaml

tests:
  - it: can enable the service account with a custom name
    templates:
      - templates/serviceaccount.yaml
    set:
      serviceAccount:
        create: true
        name: "custom-service-account"
    asserts:
      - matchSnapshot: {}

  - it: can use an existing service account
    set:
      serviceAccount:
        create: false
        existingName: "existing-service-account"
    templates:
      - templates/statefulset.yaml
    asserts:
      - equal:
          path: spec.template.spec.serviceAccountName
          value: existing-service-account

  - it: prioritizes existing service account when both create and existingName are set
    set:
      serviceAccount:
        create: true
        existingName: "existing-service-account"
        name: ""
    templates:
      - templates/statefulset.yaml
    asserts:
      - equal:
          path: spec.template.spec.serviceAccountName
          value: existing-service-account
