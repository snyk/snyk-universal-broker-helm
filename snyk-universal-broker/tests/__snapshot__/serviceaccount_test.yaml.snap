can enable the service account with a custom name:
  1: |
    apiVersion: v1
    kind: serviceaccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: snyk-universal-broker
        helm.sh/chart: snyk-universal-broker-0.0.1
      name: custom-service-account
      namespace: NAMESPACE
